src/Codeception/Lib/GroupManager.php

```diff
@@ -171,10 +173,9 @@ public function groupsForTest(\PHPUnit\Framework\Test $test)
                 }
                 if ($test instanceof \PHPUnit\Framework\TestSuite\DataProvider) {
                     $firstTest = $test->testAt(0);
-                    if ($firstTest != false && $firstTest instanceof TestInterface) {
-                        if (strpos($filename . ':' . $firstTest->getName(false), $testPattern) === 0) {
-                            $groups[] = $group;
-                        }
+                    $firstTestName = $firstTest->getName(false);
+                    if ($firstTest != false && $firstTest instanceof TestInterface && strpos($filename . ':' . $firstTestName, (string) $testPattern) === 0) {
```

Comment:
Potential bug here, `getName` method is called before checking that $firstTest is not false.